File: frontend/src/types/analysis.ts
export type Language = "python" | "cpp" | "java";
export type ResultStatus = "detected" | "ai_inferred" | "estimated" | "unknown" | "not_measured";
export type Severity = "low" | "medium" | "high";

export interface Finding {
  type: string;
  severity: Severity;
  confidence: ResultStatus;
  location?: string | null;
  description: string;
  evidence?: string | null;
}

export interface ComplexityEstimate {
  value: string;
  status: ResultStatus;
}

export interface Complexity {
  time: ComplexityEstimate;
  space: ComplexityEstimate;
}

export interface Suggestion {
  issue: string;
  why_it_matters: string;
  suggested_improvement: string;
  expected_effect: string;
  status: ResultStatus;
}

export interface AnalyzeResponse {
  summary: string;
  detected_findings: Finding[];
  inferred_findings: Finding[];
  complexity: Complexity;
  suggestions: Suggestion[];
  optimized_code: string | null;
  optimized_complexity: Complexity | null;
  optimization_explanation: string | null;
  comparison: string[];
  assumptions: string[];
  limitations: string[];
}

export interface AnalyzeRequest {
  language: Language;
  code: string;
}

File: backend/app/schemas/analysis.py
from enum import Enum
from typing import List, Optional
from pydantic import BaseModel, Field


class Language(str, Enum):
    python = "python"
    cpp = "cpp"
    java = "java"


class ResultStatus(str, Enum):
    detected = "detected"
    ai_inferred = "ai_inferred"
    estimated = "estimated"
    unknown = "unknown"
    not_measured = "not_measured"


class Severity(str, Enum):
    low = "low"
    medium = "medium"
    high = "high"


class AnalyzeRequest(BaseModel):
    language: Language
    code: str = Field(..., min_length=1)


class Finding(BaseModel):
    type: str
    severity: Severity
    confidence: ResultStatus
    location: Optional[str] = None
    description: str
    evidence: Optional[str] = None


class ComplexityEstimate(BaseModel):
    value: str
    status: ResultStatus


class Complexity(BaseModel):
    time: ComplexityEstimate
    space: ComplexityEstimate


class Suggestion(BaseModel):
    issue: str
    why_it_matters: str
    suggested_improvement: str
    expected_effect: str
    status: ResultStatus


class AnalyzeResponse(BaseModel):
    summary: str
    detected_findings: List[Finding] = []
    inferred_findings: List[Finding] = []
    complexity: Complexity
    suggestions: List[Suggestion] = []
    optimized_code: Optional[str] = None
    optimized_complexity: Optional[Complexity] = None
    optimization_explanation: Optional[str] = None
    comparison: List[str] = []
    assumptions: List[str] = []
    limitations: List[str] = []